kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala

```diff
@@ -288,6 +288,19 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
     ThreadUtils.shutdown(execPool, Duration(shutdownTimeout, TimeUnit.MILLISECONDS))
   }
 
+  protected def checkSessionTimeout(current: Long, session: Session, timeout: Long): Unit = {
+    if (session.lastAccessTime + timeout <= current && session.getNoOperationTime > timeout) {
+      try {
+        closeSession(session.handle)
+      } catch {
+        case e: KyuubiSQLException =>
+          warn(s"Error closing idle session ${session.handle}", e)
+      }
+    } else {
+      session.closeExpiredOperations()
+    }
+  }
+
   private def startTimeoutChecker(): Unit = {
     val interval = conf.get(SESSION_CHECK_INTERVAL)
     val timeout = conf.get(SESSION_IDLE_TIMEOUT)
```

Comment:
How about add a new filed `sessionTimeout` in `Session`. Use  `SESSION_IDLE_TIMEOUT ` by default in `AbstractSession`, and use `BATCH_SESSION_IDLE_TIMEOUT` in `KyuubiBatchSessionImpl`.

Then we can make each session use it's own timeout threshold.

Reply:
thanks, will address it